logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/legalhold/FetchLegalHoldForSelfUserFromRemoteUseCase.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.legalhold
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.conversation.LegalHoldStatus
+import com.wire.kalium.logic.data.id.SelfTeamIdProvider
+import com.wire.kalium.logic.data.team.TeamRepository
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.flatMap
+
+/**
+ * Use case that allows to fetch and persist the legal hold state for the self user.
+ */
+interface FetchLegalHoldForSelfUserFromRemoteUseCase {
+    suspend operator fun invoke(): Either<CoreFailure, LegalHoldStatus>
+}
+
+class FetchLegalHoldForSelfUserFromRemoteUseCaseImpl internal constructor(
+    private val teamRepository: TeamRepository,
+    private val selfTeamIdProvider: SelfTeamIdProvider,
+) : FetchLegalHoldForSelfUserFromRemoteUseCase {
+
+    override suspend fun invoke(): Either<CoreFailure, LegalHoldStatus> =
+        selfTeamIdProvider()
+            .flatMap { teamId ->
+                if (teamId == null) Either.Right(LegalHoldStatus.NO_CONSENT)
+                else teamRepository.fetchLegalHoldStatus(teamId)
```

Comment:
Shouldn't we persist the result as well ?

Reply:
`teamRepository. fetchLegalHoldStatus` uses `legalHoldHandler` or `legalHoldRequestHandler` according to actual state and these handlers are responsible for handling this event (so for persisting it and in future also creating proper system messages if needed)